### PR TITLE
Upgrade template

### PR DESCRIPTION
**Pull request by pacrob**
### What was wrong?

Python project template updates, notably adding py312 support.
Test repo for adding [blocklint](https://github.com/PrincetonUniversity/blocklint) to our linting lineup.

### Todo:

- [x] Clean up commit history

- [x] Add or update documentation related to these changes

- [x] Add entry to the [release notes](https://github.com/ethereum/eth-typing/blob/main/newsfragments/README.md)

#### Cute Animal Picture

image

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,9 +1,9 @@
 [tox]
 envlist=
-    py{38,39,310,311}-core
-    py{38,39,310,311}-lint
-    py{38,39,310,311}-wheel
-    py311-wheel-windows
+    py{38,39,310,311,312}-core
+    py{38,39,310,311,312}-lint
+    py{38,39,310,311,312}-wheel
+    windows-wheel
```
Comment (kclowes) [inline tox.ini:6]: maybe:
```suggestion
    py{311,312}-windows-wheel
```
for consistency?

**Review by kclowes — APPROVED** [review]
left a nit, but looks good otherwise!

File: tox.ini (hunk 1 above)
Comment (pacrob) [reply tox.ini:6]: As it is, it just keeps circleci config drier. See .circleci/config.yml -> windows-wheel-steps -> windows-wheel-setup. If specified different TOXENVs, the whole windows-wheel-setup step would need to be duplicated for each job. Happy to make the change if you think it helps with clarity.

Comment (kclowes) [reply tox.ini:6]: Nah, I think it's good as-is. ✨

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -48,3 +48,8 @@ repos:
     hooks:
     -   id: mypy
         exclude: tests/
+-   repo: https://github.com/PrincetonUniversity/blocklint
+    rev: v0.2.4
+    hooks:
+    -   id: blocklint
+        exclude: 'docs/Makefile|docs/release_notes.rst|tox.ini'
```
Comment (reedsa) [inline .pre-commit-config.yaml:55]: Does this find all other files in the repo? I think I was trying with listing directories to include but if this works I like it better.

Comment (pacrob) [reply .pre-commit-config.yaml:55]: I've tried sticking blocked words in various files and it hasn't missed one yet.

**Pull request merged**